### PR TITLE
Add command line interface and Google Colab description

### LINKED ISSUE #7: Write the method
- [x] General overview
- [x] Processors
- [ ] Implementation
- [ ] Experimental Setup

### PR DESCRIPTION
**Pull request by LeviUniAcc**
Add some information on how we executed our pipelines.

### CODE REVIEW

File: paper.tex [hunk 1]
```diff
@@ -117,6 +117,10 @@ \section{Method}
 
 As for text processors, we evaluated three different methods of augmenting generated texts. The first approach was to replace every white-space in a given text by two white-spaces. The second approach was to add some typos to the texts. This has been implemented based on the nlpaug library, which is capable of simulating certain typos as if they were made on a keyboard by a human. The typos include mistakes, like hitting a wrong key, which is close to the right one. This results in typos often created by humans. The last approach is a translator, which has been used to translate the given text into a certain language (in this case French) and re-translate it into english. As translators are not deterministic but aim to keep the semantic meaning, the resulting text should be semantically the same, but might have a different formulation.
 
+For ease of use and low time consumption we implemented a simple python command line interface, that makes it easy to define a pipeline consisting of a generator, processor and evaluator. It is also possible to enter multiple processors for one pipeline that are then stacked and applied together on the images.
```
Comment (EhrenbruderRoschtbraten) [inline paper.tex]:
```suggestion
For the ease of use and low time consumption we implemented a simple python command line interface, that makes it easy to define a pipeline consisting of a generator, processor and evaluator. It is also possible to enter multiple processors for one pipeline, to stack and apply them together on the content.
```

File: paper.tex [hunk 2]
```diff
@@ -117,6 +117,10 @@ \section{Method}
...
+
+The resulting pipelines especially for the images were hardware expensive and could not be runn efficiently on our own machines. We chose to use the free google collab recources, consisting of a NVIDIA Tesla T4 GPU that would speed up the pipelines up to 40x compared to our local execution time.
```
Comment (EhrenbruderRoschtbraten) [inline paper.tex]:
```suggestion
The resulting pipelines, especially for the images, were very resource intensive and could not be runned efficiently on our own machines. We chose to use the free google collab recources, consisting of a NVIDIA Tesla T4 GPU that would speed up the pipelines up to 40x compared to our local execution time. 
```

**Review by EhrenbruderRoschtbraten — COMMENTED** [review]
Looks good, only some small comments

File: paper.tex [hunk 3]
```diff
@@ -117,6 +117,10 @@ \section{Method}
...
 
+For the ease of use and low time consumption we implemented a simple python command line interface, that makes it easy to define a pipeline consisting of a generator, processor and evaluator. It is also possible to enter multiple processors for one pipeline, to stack and apply them together on the content.
+
+The resulting pipelines, especially for the images, were very resource intensive and could not be runned efficiently on our own machines. We chose to use the free google collab recources, consisting of a NVIDIA Tesla T4 GPU that would speed up the pipelines up to 40x compared to our local execution time.
```
Comment (RaphaW) [inline paper.tex]:
```suggestion
The resulting pipelines, especially for the images, were very resource intensive and could not be runned efficiently on our own machines. We chose to use the free Google Colab recources, consisting of a NVIDIA Tesla T4 GPU that would speed up the pipelines up to 40x compared to our local execution time. 
```

**Review by RaphaW — APPROVED** [review]
One minor adjustment

File: paper.tex [hunk 4]
```diff
@@ -117,6 +117,10 @@ \section{Method}
 
 As for text processors, we evaluated three different methods of augmenting generated texts. The first approach was to replace every white-space in a given text by two white-spaces. The second approach was to add some typos to the texts. This has been implemented based on the nlpaug library, which is capable of simulating certain typos as if they were made on a keyboard by a human. The typos include mistakes, like hitting a wrong key, which is close to the right one. This results in typos often created by humans. The last approach is a translator, which has been used to translate the given text into a certain language (in this case French) and re-translate it into english. As translators are not deterministic but aim to keep the semantic meaning, the resulting text should be semantically the same, but might have a different formulation.
 
+For the ease of use and low time consumption we implemented a simple python command line interface, that makes it easy to define a pipeline consisting of a generator, processor and evaluator. It is also possible to enter multiple processors for one pipeline, to stack and apply them together on the content.
```
Comment (1Maxnet1) [inline paper.tex]: I would add something like the following. The point I find is missing, is the fact that the software allows us to easily manage experiments, as we have to enter one command to run a hole experiment. 
```suggestion
For the ease of use, low time consumption and easy management of our experiments we implemented a simple python command line interface, that allows defining pipelines consisting of a generator, processor and evaluator. It is also possible to enter multiple processors for one pipeline to stack and apply them together on the content.
```

File: paper.tex (hunk 3 above)
Comment (1Maxnet1) [inline paper.tex]: Fixed two additional typos. Also I would add the aspect of having a consistent environment for the different environments.
```suggestion
The resulting pipelines, especially for the images, were very resource intensive and could not be run efficiently on our own machines. We chose to use the free Google Colab resources, consisting of a NVIDIA Tesla T4 GPU that would speed up the pipelines up to 40x compared to our local execution time and allowed to have the same environment for all experiments. 
```

**Review by 1Maxnet1 — COMMENTED** [review]
Thanks for the paragraphs, left two comments.

**Review by EhrenbruderRoschtbraten — CHANGES_REQUESTED** [review]
Made two small adjustements to @1Maxnet1 comment

File: paper.tex (hunk 3 above)
Comment (LeviUniAcc) [reply paper.tex]: change in maxnets request

**Review by 1Maxnet1 — APPROVED** [review]
LGTM

File: paper.tex (hunk 4 above)
Comment (EhrenbruderRoschtbraten) [inline paper.tex]:
```suggestion
For the ease of use and low time consumption we implemented a simple python command line interface, that makes it easy to define a pipeline consisting of a generator, processor and evaluator. It is also possible to enter multiple processors for one pipeline to stack and apply them together on the content.
```

File: paper.tex [hunk 5]
```diff
@@ -117,6 +117,10 @@ \section{Method}
...
+
+The resulting pipelines, especially for the images, were very resource intensive and could not be run efficiently on our own machines. We chose to use the free Google Colab resources, consisting of a NVIDIA Tesla T4 GPU that would speed up the pipelines up to 40x compared to our local execution time and allowed to have the same environment for all experiments.
```
Comment (EhrenbruderRoschtbraten) [inline paper.tex]:
```suggestion
The resulting pipelines, especially for images, were very resource intensive and could not be run efficiently on our own machines. We chose to use the free Google Colab resources, consisting of a NVIDIA Tesla T4 GPU that speeds up the pipelines up to 40x compared to our local execution time and allowed us to use the same environment for all experiments. 
```

**Review by EhrenbruderRoschtbraten — COMMENTED** [review]
small comments

**Review by EhrenbruderRoschtbraten — APPROVED** [review]
Wird schon passen denk

**Pull request merged**